packages/phishing-controller/src/PhishingController.test.ts

```diff
@@ -25,213 +26,393 @@ describe('PhishingController', () => {
+      expect(controller.isStalelistOutOfDate()).toBe(false);
+      clock.tick(1000 * 10);
+      expect(controller.isStalelistOutOfDate()).toBe(true);
       expect(nockScope.isDone()).toBe(true);
     });
 
     it('should call update only if it is out of date, otherwise it should not call update', async () => {
       const clock = sinon.useFakeTimers();
-      const controller = new PhishingController({ refreshInterval: 10 });
-      expect(controller.isOutOfDate()).toBe(false);
-      await controller.maybeUpdatePhishingLists();
+      const controller = new PhishingController({
+        stalelistRefreshInterval: 10,
+      });
+      expect(controller.isStalelistOutOfDate()).toBe(false);
+      await controller.maybeUpdateState();
       expect(
-        controller.test('this-should-not-be-in-default-blacklist.com'),
+        controller.test('this-should-not-be-in-default-blocklist.com'),
       ).toMatchObject({
         result: false,
         type: 'all',
       });
 
       expect(
-        controller.test('this-should-not-be-in-default-whitelist.com'),
+        controller.test('this-should-not-be-in-default-allowlist.com'),
       ).toMatchObject({
         result: false,
         type: 'all',
       });
 
-      clock.tick(10);
-      await controller.maybeUpdatePhishingLists();
+      clock.tick(1000 * 10);
+      await controller.maybeUpdateState();
 
       expect(
-        controller.test('this-should-not-be-in-default-blacklist.com'),
+        controller.test('this-should-not-be-in-default-blocklist.com'),
       ).toMatchObject({
         result: true,
         type: 'blocklist',
       });
 
       expect(
-        controller.test('this-should-not-be-in-default-whitelist.com'),
+        controller.test('this-should-not-be-in-default-allowlist.com'),
       ).toMatchObject({
         result: false,
         type: 'allowlist',
       });
 
       expect(nockScope.isDone()).toBe(true);
     });
+
+    it('should not have hotlist be out of date immediately after maybeUpdateState is called', async () => {
+      nockScope = nock(PHISHING_CONFIG_BASE_URL)
+        .get(METAMASK_HOTLIST_DIFF_FILE)
+        .reply(200, [
+          {
+            url: 'this-should-not-be-in-default-blocklist.com',
+            timestamp: 1,
+            isRemoval: true,
+            targetList: 'blocklist',
+          },
+          {
+            url: 'this-should-not-be-in-default-blocklist.com',
+            timestamp: 2,
+            targetList: 'blocklist',
+          },
+        ]);
+      const clock = sinon.useFakeTimers(50);
+      const controller = new PhishingController({
+        hotlistRefreshInterval: 10,
+        stalelistRefreshInterval: 50,
+      });
+      clock.tick(1000 * 10);
+      expect(controller.isHotlistOutOfDate()).toBe(true);
+      await controller.maybeUpdateState();
+      expect(controller.isHotlistOutOfDate()).toBe(false);
+      // expect(nockScope.isDone()).toBe(true);
```

Comment:
Refactoring this to avoid the unused network mock would be ideal, but the test doesn't strictly need this assertion, so deleting this comment instead would be OK for now.